src/v/cluster/topics_frontend.cc

```diff
@@ -1391,4 +1397,118 @@ ss::future<std::error_code> topics_frontend::move_partition_replicas(
+            return ss::make_ready_future<result<partition_state_reply>>(reply);
+        }
+        return _pm.invoke_on(
+          *shard,
+          [ntp = std::move(ntp),
+           reply = std::move(reply)](partition_manager& pm) mutable {
+              auto partition = pm.get(ntp);
+              if (!partition) {
+                  reply.error_code = errc::partition_not_exists;
+                  return ss::make_ready_future<result<partition_state_reply>>(
+                    reply);
+              }
+              reply.state = ::cluster::get_partition_state(partition);
+              reply.error_code = errc::success;
+              return ss::make_ready_future<result<partition_state_reply>>(
+                reply);
+          });
+    }
+    auto timeout = model::timeout_clock::now() + 5s;
+    return _connections.local().with_node_client<controller_client_protocol>(
+      _self,
+      ss::this_shard_id(),
+      node,
+      timeout,
+      [ntp = std::move(ntp),
+       timeout](controller_client_protocol client) mutable {
+          return client
+            .get_partition_state(
+              partition_state_request{.ntp = ntp}, rpc::client_opts(timeout))
+            .then(&rpc::get_ctx_data<partition_state_reply>);
+      });
+}
+
+ss::future<result<std::vector<partition_state>>>
+topics_frontend::get_partition_state(model::ntp ntp) {
+    const auto& topics = _topics.local();
+    if (!topics.contains(model::topic_namespace_view(ntp))) {
+        co_return errc::partition_not_exists;
+    }
+    std::set<model::node_id> nodes_to_query;
+    const auto& current = topics.get_partition_assignment(ntp);
+    if (current) {
+        const auto& bss = current->replicas;
+        std::for_each(
+          bss.begin(),
+          bss.end(),
+          [&nodes_to_query](const model::broker_shard& bs) {
+              nodes_to_query.insert(bs.node_id);
+          });
+    }
+    const auto& prev = topics.get_previous_replica_set(ntp);
+    if (prev) {
+        std::for_each(
+          prev.value().begin(),
+          prev.value().end(),
+          [&nodes_to_query](const model::broker_shard& bs) {
+              nodes_to_query.insert(bs.node_id);
+          });
+    }
+
+    if (nodes_to_query.empty()) {
+        co_return errc::no_partition_assignments;
+    }
+
+    std::vector<ss::future<result<partition_state_reply>>> futures;
+    futures.reserve(nodes_to_query.size());
+    for (const auto& node : nodes_to_query) {
+        futures.push_back(do_get_partition_state(node, ntp));
+    }
+
+    auto finished_futures = co_await ss::when_all(
+      futures.begin(), futures.end());
+    std::vector<partition_state> results;
+    results.reserve(finished_futures.size());
+    for (auto& fut : finished_futures) {
+        if (fut.failed()) {
+            vlog(
+              clusterlog.debug,
+              "Failed to get partition state for ntp: {}, failure: {}",
+              ntp,
+              fut.get_exception());
```

Comment:
it's a minor thing, but can you pull this out of the log statement like:

```
auto ex = fut.get_exception();
vlog(..., ex);
```

The reason is that `get_exception` has an important side-effect (not throwing abandoned exceptional future) and I think it better to be a bit clearer to anyone in the future who might refactor something.

Reply:
Done.